version: '3'
services:
  redis:
    image: redis:4.0.1
    container_name: redis
    volumes:
      - /home/ubuntu/chatgpt-telegram-bot-py/redis/data:/data
      - /home/ubuntu/chatgpt-telegram-bot-py/redis/conf/redis.conf:/usr/local/etc/redis/redis.conf
      - /home/ubuntu/chatgpt-telegram-bot-py/redis/logs:/logs
    command: redis-server --requirepass frM991103
    networks:
      - my_network
    ports:
      - 6379:6379
    
  chatbot:
    image: chatbot:latest
    container_name: chatbot
    volumes:
      - /home/ubuntu/chatgpt-telegram-bot-py:/chatgpt-telegram-bot-py
    command: python3 main.py
    networks:
      - my_network
    depends_on:
      - redis

  app:
    image: chatbot:latest
    container_name: app
    volumes:
      - /home/ubuntu/chatgpt-telegram-bot-py:/chatgpt-telegram-bot-py
    command: python3 app.py
    networks:
      - my_network
    depends_on:
      - redis
      - chatbot
    ports:
      - 8000:8000


networks:
  my_network: